standup/auth0/middleware.py

```diff
@@ -44,78 +45,84 @@ def renew_id_token(id_token):
     return result.get('id_token')
 
 
-def get_path(url):
-    """Takes a url and returns path + querystring"""
-    parsed = urlparse(url)
-    if parsed.query:
-        return '%s?%s' % (parsed.path, parsed.query)
-    return parsed.path
+class ValidateIdToken(object):
+    """For users authenticated with an id_token, we need to check that it's still valid. For
+    example, the user could have been blocked (e.g. leaving the company) if so we need to ask the
+    user to log in again.
 
+    We do this using cache.
 
-class ValidateIDToken(object):
-    """For users authenticated with an id_token, we need to check that it's still valid. For example,
-    the user could have been blocked (e.g. leaving the company) if so we need to ask the user to log
-    in again.
+    # FIXME(willkg): Rework this so that the cache parts are methods that can be overridden.
 
     """
 
     exception_paths = (
-        get_path(settings.AUTH0_CALLBACK),
+        # Exclude the AUTH0_CALLBACK_URL path, otherwise this can loop
+        urlparse(settings.AUTH0_CALLBACK_URL).path,
     )
 
     def process_request(self, request):
         if (
-            request.method != 'POST' and
-            not request.is_ajax() and
-            request.user.is_active and
-            request.path not in self.exception_paths
+                request.method != 'POST' and
+                # FIXME(willkg): We might want to do this for AJAX, too, otherwise one-page webapps
+                # might never renew.
+                not request.is_ajax() and
+                request.user.is_active and
+                request.path not in self.exception_paths
         ):
-            # Look up expiration in session and see if the id_token needs to be renewed.
-            id_token_expiration = request.session.get('id_token_expiration', None)
-            if id_token_expiration and id_token_expiration < datetime.utcnow():
+            cache_key = 'auth0:renew_id_token:%s' % request.user.id
+
+            # Look up expiration in cache to see if id_token needs to be renewed.
+            if cache.get(cache_key):
```

Comment:
Since we're using multiple processes and dummy cache, this probably won't work for us. However, if we don't stash it here, I'm not sure where else to put it other than in the db. If we use the db, that means we're doing another db hit for every request. Though maybe that's fine for sites that don't have memcache.

We don't want it in the session or in a cookie because it dictates when we try to renew this token and thus it should never get to the client.